### ISSUE TITLE
makecpt crashes if -T picks only 1 point

### ISSUE THREAD

**Issue by PaulWessel** [GenericMappingTools/gmt#5016]
This crashes:

`gmt makecpt -Cjet -Z -T0/100/1+n`

The problem was we were trying to set the previous color to the last one but if there is only one...
So I am/can fix that so it no longer crashes.  However, this gives this output:

```
gmt makecpt -Cjet -Z -T0/100/1+n
B	black
F	white
N	128

```
Because -T..+n sets the number of color boundaries.  I guess with 1+n I would expect in this case to get this:

`0	0/0/127	100	127/0/0	B`	

which is what we get for 2+n.  I.e., I am interpreting -T0/100/1+n to mean "give me one slice" which then technically is the same as -T0/100/2+n or -T0/100/100.  Are you OK with that fix so we dont get a non-existant CPT on the output, @joa-quim @meghanrjones  @seisman ?

**Comment by meghanrjones** [comment]
I'm a bit concerned about whether users would have based their code off the documentation below in the [generate 1D array section](https://docs.generic-mapping-tools.org/dev/filter1d.html#generate-1d-array) (which I think are the only docs for **-T+n**), which accurately describes the current behavior (i.e., ``-T0/100/2+n`` is needed here):
<img width="928" alt="image" src="https://user-images.githubusercontent.com/14077947/112397163-eef7b300-8cd7-11eb-89c5-abd3240caee2.png">

**Comment by meghanrjones** [comment]
Would this change only affect ``makecpt`` or all modules that use **-T**?

**Comment by PaulWessel** [comment]
This would be makecpt only since in gmtmath we want the current behaviour:

```
gmt math -T0/100/1+n T =
100	100
```

Another option is to simply give an error and say at least 2 nodes are required to form a CPT slice.

**Comment by meghanrjones** [comment]
> This would be makecpt only since in gmtmath we want the current behaviour:
> 
> ```
> gmt math -T0/100/1+n T =
> 100	100
> ```
> 
> Another option is to simply give an error and say at least 2 nodes are required to form a CPT slice.

I prefer the error option, in case people have already adjusted for this bug, but am fine with either.

**Comment by PaulWessel** [comment]
I am coming around to believe that we should simpy give an error for +n1 since 2 is the minimum to form a slice.

**Comment by PaulWessel** [comment]
OK, just say your comment after I agreed with your future comment.

**Comment by PaulWessel** [comment]
Pr to come now does this:

gmt makecpt -Cjet -Z -T0/100/1+n
makecpt [ERROR]: Need at least two nodes to form a CPT slice, your choice only resulted in 1 nodes

**Comment by meghanrjones** [comment]
Closed by  #5018
